uplink/piecestore/upload.go

```diff
@@ -67,7 +67,11 @@ func (client *Client) Upload(ctx context.Context, limit *pb.OrderLimit, piecePri
 	})
 	if err != nil {
 		_, closeErr := stream.CloseAndRecv()
-		return nil, ErrProtocol.Wrap(errs.Combine(err, closeErr))
+		if err != io.EOF {
```

Comment:
If `err == io.EOF` and `closeErr == nil` then this wouldn't return `nil` as the error.

Reply:
doesn't `Wrap` return nil if the passed err is nil?

Reply:
Ah, sorry, I meant it would return `nil`... the same for the other. I.e. it should always return some error, but adding this special case means it might skip it.

Reply:
oh, you mean that in case that `err == io.EOF` and `closeErr == nil` we should return `io.EOF`?

Reply:
my next comment went after yours, I expected that was going to be sent before your reply.

I'll change to return `io.EOF`